src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -149,10 +149,10 @@ fsm_init_primary(Keeper *keeper)
 		/*
 		 * We have a new system_identifier, we need to publish it now.
 		 */
-		if (!monitor_set_node_system_identifier(
-				monitor,
-				keeper->state.current_node_id,
-				pgSetup->control.system_identifier))
+		if (!config->monitorDisabled &&
```

Comment:
I'm on the fence to saying that I prefer having the monitorDisabled test in its own statement, even when everything is indented one level more, to make it very obvious when reading the code... it's nitpicking at this point though.

Reply:
Done.